Simplify deployment code and documentation

fregante opened the pull request:
Replaces #2 
Should fix current deployment issue

1. Starts by describing local deployment (easiest)
2. Then moves onto the automated deployment on Travis
3. Then includes simple instructions to use tags.

fregante committed:
Update cronjob deployment

fregante committed:
Replace versioning script

fregante committed:
Restore time-based condition

fregante committed:
Update readme.md

fregante committed:
Update readme.md

notlmn commented on .travis.yml:
```diff
@@ -2,22 +2,14 @@ language: node_js
-      #   On Git tags
-      tags: true
-      condition: $RELEASE_ON_TAGS = true
+      condition: (($TRAVIS_EVENT_TYPE = api) || ($TRAVIS_EVENT_TYPE = cron && $(git rev-list -n 1 --since=\"26 hours ago\" master)))
```
Drop quotes around "26 hours ago", we originally needed quotes in RGH as it was inside package.json.

fregante replied:
I think that’s needed, I think I would have dropped it in RGH otherwise

notlmn commented on readme.md:
````diff
@@ -147,33 +146,51 @@ The options page includes [fregante/webext-options-sync][link-options-sync] that
-And don't forget to setup a cron job that runs daily on master, and include your CWS extension id in [.travis.yml](.travis.yml).
+And then running:
+
+```sh
````
```suggestion
``` sh
```

fregante replied:
?

notlmn replied:
I always like to have a single space before the language specifier. It looks a bit aligned at 4 positions from the left.

The rest of the code also has a single space before the language specifier.

fregante replied:
I’ve never seen that before but apparently it’s allowed by CommonMark https://spec.commonmark.org/0.29/#example-113 and also works on StackOverflow 🤷‍♂️

notlmn replied:
Yup, I'm aware that works with/without spaces. But I like to keep it spaced, a convention that I follow in Markdown.

notlmn commented on readme.md:
```diff
@@ -147,33 +146,51 @@ The options page includes [fregante/webext-options-sync][link-options-sync] that
+
+This will:
+
+1. Build the extension
```
Changes these list numbers be all `1`s instead of `1` `2` `3`, to be consistent.

filips123 replied:
Wouldn't be better to use `1` `2` `3` at all places?

notlmn replied:
It would be harder to reorder and add/delete new list items, as it requires us to change all most of them.

notlmn commented on readme.md:
````diff
@@ -147,33 +146,51 @@ The options page includes [fregante/webext-options-sync][link-options-sync] that
+```
 
-To set it up, set the environment variable `RELEASE_ON_TAGS` to `true` in [.travis.yml](.travis.yml). In this case, [automatic daily releases](#auto-publishing) and its respective [versioning](#versioning) method will not be triggered.
+And your extension will be published when creating a git tag, using the tag itself as version for the manifest.
````
```suggestion
And your extension will be published when creating a git tag, using the tag itself as version for the extension.
```

This sounds a bit nice?

fregante committed:
Update readme.md

Co-Authored-By: Laxman <notlmn@outlook.com>

fregante committed:
Update readme.md

Co-Authored-By: Laxman <notlmn@outlook.com>

filips123 commented:
Why not keep using `RELEASE_ON_TAGS` environment variable? Travis should detect it and run deploy only on Tags. If this is broken, it would be better to just fix condition and update docs.
It would also be better for users because they would only need to change one variable instead of complete deployment code.

fregante commented:
Because this is not a package, it’s code that the users will fork and maintain. It should be as simple as possible *for them*. The more you can simplify it, the better it is for the user.

fregante committed:
Improve .travis.yml readability

fregante committed:
Shorten deployment condition

yesterday = 24 hours ago
https://github.com/git/git/blob/3034dab9ed6b11970a53099a7b3ca981f1461365/date.c#L1034

fregante committed:
Prefer the tag as SOURCE_URL when available

fregante committed:
Markdown style prefs

notlmn commented:
LGTM. Merge if you can verify that the deployment script actually works. 😄

fregante commented on .travis.yml:
```diff
@@ -1,23 +1,15 @@
-      #   On Git tags
-      tags: true
-      condition: $RELEASE_ON_TAGS = true
+      condition: (($TRAVIS_EVENT_TYPE = api) || ($TRAVIS_EVENT_TYPE = cron && $(git rev-list --since=yesterday HEAD)))
```
Condition verified locally with:

``` sh
if [[ condition ]]; then echo 1; fi;
```
``` sh
TRAVIS_EVENT_TYPE=api
if [[ condition ]]; then echo 1; fi;
```
``` sh
TRAVIS_EVENT_TYPE=cron
if [[ condition ]]; then echo 1; fi;
```

fregante commented on package.json:
```diff
@@ -8,8 +8,8 @@
-		"prerelease:version": "if [[ $RELEASE_ON_TAGS = true ]]; then VERSION=$(echo $TRAVIS_TAG); else VERSION=$(utc-version); fi; echo $VERSION; dot-json distribution/manifest.json version $VERSION",
-		"prerelease:source-url": "echo https://github.com/$TRAVIS_REPO_SLUG/tree/\"${TRAVIS_COMMIT:-$VERSION}\" > distribution/SOURCE_URL",
+		"prerelease:version": "VERSION=$(utc-version); echo $VERSION; dot-json distribution/manifest.json version $VERSION",
+		"prerelease:source-url": "if [[ $TRAVIS_REPO_SLUG ]]; echo https://github.com/$TRAVIS_REPO_SLUG/tree/\"${TRAVIS_TAG:-$TRAVIS_COMMIT}\" > distribution/SOURCE_URL; fi",
```
```suggestion
		"prerelease:source-url": "if [[ $TRAVIS_REPO_SLUG ]]; then echo https://github.com/$TRAVIS_REPO_SLUG/tree/\"${TRAVIS_TAG:-$TRAVIS_COMMIT}\" > distribution/SOURCE_URL; fi",
```

Verified with:

```sh
if [[ $TRAVIS_REPO_SLUG ]]; then echo https://github.com/$TRAVIS_REPO_SLUG/tree/"${TRAVIS_TAG:-$TRAVIS_COMMIT}"; fi
# no output

TRAVIS_REPO_SLUG=notlmn/browser
TRAVIS_COMMIT=abcd123456
if [[ $TRAVIS_REPO_SLUG ]]; then echo https://github.com/$TRAVIS_REPO_SLUG/tree/"${TRAVIS_TAG:-$TRAVIS_COMMIT}"; fi
# output: https://github.com/notlmn/browser/tree/abcd123456

TRAVIS_TAG=1.2.3
if [[ $TRAVIS_REPO_SLUG ]]; then echo https://github.com/$TRAVIS_REPO_SLUG/tree/"${TRAVIS_TAG:-$TRAVIS_COMMIT}"; fi
# output: https://github.com/notlmn/browser/tree/1.2.3
```

fregante reviewed (commented):
Verified

fregante commented:
> LGTM. Merge if you can verify that the deployment script actually works. 😄

To verify it actually works: https://github.com/notlmn/browser-extension-template/issues/11#issuecomment-515394709

fregante committed:
Fix prerelease script

fregante commented:
<img width="565" alt="Screenshot 2019-07-26 at 20 12 56" src="https://user-images.githubusercontent.com/1402241/61953909-c5a5f180-afe1-11e9-9695-e40377814af8.png">

1s response time
